File: February/229. Majority Element II.java
// 229. Majority Element II
// https://leetcode.com/problems/majority-element-ii/

class Solution {
    public List<Integer> majorityElement(int[] nums) {
        int n = nums.length;
        Map<Integer, Integer> m = new HashMap<>();
        for (int i : nums) {
            m.put(i, m.getOrDefault(i, 0) + 1);
        }
        List<Integer> ans = new ArrayList<>();
        for (Map.Entry<Integer, Integer> entry : m.entrySet()) {
            if (entry.getValue() > n / 3) {
                ans.add(entry.getKey());
            }
        }
        return ans;
    }
}

File: February/229. Majority Element II.py
# 229. Majority Element II
# https://leetcode.com/problems/majority-element-ii/

class Solution(object):
    def majorityElement(self, nums):
        n = len(nums)
        m = {}
        for i in nums:
            if i in m:
                m[i] += 1
            else:
                m[i] = 1
        ans = []
        for key, value in m.items():
            if value > n // 3:
                ans.append(key)
        return ans
